core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpFunctionImpl.java

```diff
@@ -78,23 +85,42 @@
                 parameters[i] = ctx;
             } else if (entry.getType().isAssignableFrom(HttpRequest.class)) {
                 parameters[i] = req;
+            } else if (entry.getType().isAssignableFrom(AggregatedHttpMessage.class)) {
+                parameters[i] = message;
             }
         }
         return parameters;
     }
 
     @Override
     public Object serve(ServiceRequestContext ctx, HttpRequest req, Map<String, String> args) throws Exception {
-        Object[] parameters = parameters(ctx, req, args);
+        if (requireAggregate) {
+            if (CompletionStage.class.isAssignableFrom(method.getReturnType())) {
+                return req.aggregate()
+                          .thenComposeAsync(msg -> (CompletionStage<?>) invokeMethod(ctx, req, args, msg),
```

Comment:
Thanks! 😃  Do you mean like the following? May I know I caught your point correctly?
```
        if (requireAggregate) {
            if (CompletionStage.class.isAssignableFrom(method.getReturnType())) {
                return req.aggregate()
                          .thenCompose(msg -> (CompletionStage<?>) invokeMethod(ctx, req, args, msg));
            } else if (isAsynchronous) {
                return req.aggregate()
                          .thenApply(msg -> invokeMethod(ctx, req, args, msg));
            } else {
                return req.aggregate()
                          .thenApplyAsync(msg -> invokeMethod(ctx, req, args, msg), ctx.blockingTaskExecutor());
            }
        }
```